### PR TITLE
#24 - add Link and BinBin providers

### LINKED ISSUE #24: [Providers] Add Link provider
The aim of this task is to import Link provider to the application.
Link to provider's site: https://superpedestrian.com/ (I think so)

### LINKED ISSUE #92: Add BinBin provider
Site: https://www.binbin.tech/lokasyonlar

### PR DESCRIPTION
**Pull request by AleksandraKozubal**
Added translation method in DataImporter, I think it could be helpful to solve issue #104.

It should close #24, and close #92

### CODE REVIEW

File: app/Importers/DataImporter.php
Comment (krzysztofrewak) [inline app/Importers/DataImporter.php:54]: Could you inject it to the constructor?

File: app/Importers/BinBinDataImporter.php
Comment (krzysztofrewak) [inline app/Importers/BinBinDataImporter.php:70]:
```suggestion
        }

        $this->deleteMissingProviders(self::getProviderName(), $existingCityProviders);
```

File: app/Importers/BinBinDataImporter.php
Comment (krzysztofrewak) [inline app/Importers/BinBinDataImporter.php:53]: Please avoid using strings like `"en"` directly in your code. Maybe assiging it to some constant would be better approach?

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

**Comment by krzysztofrewak** [comment]
Also: listing available providers in readme file would be a nice thing.

File: app/Importers/DataImporter.php
Comment (AleksandraKozubal) [reply app/Importers/DataImporter.php]: sure, done

File: app/Importers/BinBinDataImporter.php
Comment (AleksandraKozubal) [reply app/Importers/BinBinDataImporter.php]: added

File: app/Importers/BinBinDataImporter.php
Comment (AleksandraKozubal) [reply app/Importers/BinBinDataImporter.php]: moved to constant

File: app/Importers/DataImporter.php
Comment (AleksandraKozubal) [reply app/Importers/DataImporter.php]: It creates a problem with ImporterTest - GoogleTranslate is a final class, it can't be mocked

Comment (AleksandraKozubal) [reply app/Importers/DataImporter.php]: done

File: app/Importers/BinBinDataImporter.php
Comment (kamilpiech97) [inline app/Importers/BinBinDataImporter.php:55]:
```suggestion
            if (trim($data[1]) === "Türkiye") 
                  $countryName = "Turkey"; 
            else 
                  $countryName = $this->translate(trim($data[1]), self::language);

            if (trim($data[0]) === "Uşak") 
                  $cityName = "Usak"; 
            elseif (trim($data[0]) === "Murter") 
                  continue;
            else 
                  $cityName = $this->translate(trim($data[0]), "en");
```

Comment (kamilpiech97) [reply app/Importers/BinBinDataImporter.php:55]: more readable?

File: readme.md
Comment (krzysztofrewak) [inline readme.md:19]:
```suggestion
1. BinBin
1. Bird  
1. BitMobility
1. Bolt
1. Dott
1. Hulaj
1. Lime
1. Link
1. Neuron
1. Quick
1. Ryde
1. Spin
1. Tier
1. Urent
1. Voi
1. Zwings
```
Markdown supports numbered lists: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#lists

File: app/Importers/BinBinDataImporter.php
Comment (krzysztofrewak) [inline app/Importers/BinBinDataImporter.php:64]: This code formatting is fucked up. PHPStorm Ctrl+Alt+L should fix it, but you can also wait until this is merged: https://github.com/blumilksoftware/codestyle/pull/99

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

File: app/Importers/BinBinDataImporter.php
Comment (AleksandraKozubal) [reply app/Importers/BinBinDataImporter.php]: changed

File: app/Importers/BinBinDataImporter.php
Comment (AleksandraKozubal) [reply app/Importers/BinBinDataImporter.php]: fixed

**Review by krzysztofrewak — APPROVED** [review]

File: app/Importers/DataImporter.php
Comment (kamilpiech97) [inline app/Importers/DataImporter.php:55]:
```suggestion
    public function translate(string $word, string $language): string
```

File: readme.md
Comment (kamilpiech97) [inline readme.md:22]:
```suggestion
1. Beam
2. Beryl
3. BinBin
4. Bird
5. BitMobility
6. Bolt
7. Dott
8. Hulaj
9. Lime
10. Link
11. Neuron
12. Quick
13. Ryde
14. Spin
15. Tier
16. Urent
17. Veo
18. Voi
19. Zwings
```

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

File: readme.md
Comment (AleksandraKozubal) [reply readme.md:22]: This was suggested in https://github.com/blumilksoftware/escooters/pull/117#discussion_r1299739789

File: app/Importers/DataImporter.php
Comment (AleksandraKozubal) [reply app/Importers/DataImporter.php]: changed

**Review by kamilpiech97 — APPROVED** [review]

**Review by Letha0 — APPROVED** [review]

**Pull request merged**
